From 80cba3292b77db919a9a822d5cb3d5fa79e38be5 Mon Sep 17 00:00:00 2001
From: Benjamin Gordon <bmgordon@google.com>
Date: Wed, 2 Jul 2025 16:28:57 -0600
Subject: [PATCH] PrintSpooler: Require empty output for PDF

When saving to PDF, request a truncated output stream in case the user
selects to overwrite an existing file.

Bug: 423815728
Test: Overwrite existing PDF
Flag: EXEMPT bugfix
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:d65493c65a94415208d4be2454181f214fb2ed03)
Merged-In: Id598cd1a9d2456566f3905432eb7b5ffeab9d33e
Change-Id: Id598cd1a9d2456566f3905432eb7b5ffeab9d33e
---
 .../src/com/android/printspooler/model/RemotePrintDocument.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/packages/PrintSpooler/src/com/android/printspooler/model/RemotePrintDocument.java b/packages/PrintSpooler/src/com/android/printspooler/model/RemotePrintDocument.java
index bfc00bb8b94d..985482904727 100644
--- a/packages/PrintSpooler/src/com/android/printspooler/model/RemotePrintDocument.java
+++ b/packages/PrintSpooler/src/com/android/printspooler/model/RemotePrintDocument.java
@@ -376,7 +376,7 @@ public final class RemotePrintDocument {
         try {
             file = mDocumentInfo.fileProvider.acquireFile(null);
             in = new FileInputStream(file);
-            out = contentResolver.openOutputStream(uri);
+            out = contentResolver.openOutputStream(uri, "wt");
             final byte[] buffer = new byte[8192];
             while (true) {
                 final int readByteCount = in.read(buffer);
-- 
2.50.1.351.gcf940e82a1-goog

